### PR TITLE
Added automatic schema generation modules.

### PR DESCRIPTION
**Pull request by pmelab**
Generates a schema based on view modes and field configurations. Views integration and mutations in the pipeline.

### CODE REVIEW

File: modules/graphql_block/src/BlockExtractor.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  /**
+   * Extract the required context and return it.
+   *
+   * @return \Drupal\graphql_block\BlockResponse
+   *   A context response instance.
+   */
+  public function extract() {
+    $region = \Drupal::request()->attributes->get('graphql_block_region');
```
Comment (fubhy) [inline modules/graphql_block/src/BlockExtractor.php]: You can inject the requestStack and then do $$this->requestStack->getCurrentRequest()....

File: modules/graphql_content/src/Plugin/Deriver/FieldFormatterDeriver.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    $this->derivatives = [];
+
+    /** @var EntityTypeManagerInterface $typemanager */
+    $typemanager = \Drupal::service('entity_type.manager');
```
Comment (fubhy) [inline modules/graphql_content/src/Plugin/Deriver/FieldFormatterDeriver.php]: $this->entityTypeManager

File: modules/graphql_core/src/ContextExtractor.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  /**
+   * Extract the required context and return it.
+   *
+   * @return \Drupal\graphql_core\ContextResponse
+   *   A context response instance.
+   */
+  public function extract() {
+    $context_id = \Drupal::request()->attributes->get('graphql_context');
```
Comment (fubhy) [inline modules/graphql_core/src/ContextExtractor.php]: Same here. Let's inject the request stack.

File: modules/graphql_core/src/ContextExtractor.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    $response = new ContextResponse();
+    $response->setContext(\Drupal::service('graphql.context_repository')->getRuntimeContexts([$context_id])[$context_id]);
```
Comment (fubhy) [inline modules/graphql_core/src/ContextExtractor.php]: Same here.

**Review by fubhy — CHANGES_REQUESTED** [review]
Holy shit. This is sick @pmelab. Thanks so much :). Just a small amount of DI related remarks.

**Comment by fubhy** [comment]
Oh, and would you mind updating the main README.md to point ppl to these submodules?

File: modules/graphql_block/src/BlockExtractor.php (hunk 1 above)
Comment (pmelab) [reply modules/graphql_block/src/BlockExtractor.php]: Whoops, looks like I've been in a rush.

**Comment by pmelab** [comment]
@fubhy Done!

**Comment by fubhy** [comment]
Such a good job. Thanks a bunch @pmelab. This is going to make many people very happy.

**Pull request merged**